From e4bc7e57a6dbaeffc17f6eee0c803e9b07c404b6 Mon Sep 17 00:00:00 2001
Message-Id: <patch-4.6-e4bc7e57a6d-20210908T151929Z-avarab@gmail.com>
In-Reply-To: <patch-1.6-b5c79d87847-20210908T151929Z-avarab@gmail.com>
References: <patch-1.6-b5c79d87847-20210908T151929Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 14 Jun 2021 18:33:41 +0200
Subject: [PATCH 4/6] help: refactor "for_human" control flow in cmd_help()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Cc: Junio C Hamano <gitster@pobox.com>,
    Philip Oakley <philipoakley@iee.email>,
    Nguyễn Thái Ngọc Duy <pclouds@gmail.com>,
    SZEDER Gábor <szeder.dev@gmail.com>

Instead of having two lines that call list_config_help(for_human)
let's setup the pager and print the trailer conditionally. This makes
it clearer at a glance how the two differ in behavior.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 builtin/help.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/builtin/help.c b/builtin/help.c
index 0f9dc31c40f..0737b22069b 100644
--- a/builtin/help.c
+++ b/builtin/help.c
@@ -574,13 +574,12 @@ int cmd_help(int argc, const char **argv, const char *prefix)
 	if (show_config) {
 		int for_human = show_config == 1;
 
-		if (!for_human) {
-			list_config_help(for_human);
-			return 0;
-		}
-		setup_pager();
+		if (for_human)
+			setup_pager();
 		list_config_help(for_human);
-		printf("\n%s\n", _("'git help config' for more information"));
+		if (for_human)
+			printf("\n%s\n", _("'git help config' for more information"));
+
 		return 0;
 	}
 
-- 
2.33.0.825.g2bf60429931

